services/datamanager/builtin/builtin.go

```diff
@@ -474,11 +484,11 @@ func (svc *builtIn) uploadData(cancelCtx context.Context, intervalMins float64)
 	intervalMillis := 60000.0 * intervalMins
 	// The ticker must be created before uploadData returns to prevent race conditions between clock.Ticker and
 	// clock.Add in sync_test.go.
-	ticker := clock.Ticker(time.Millisecond * time.Duration(intervalMillis))
+	svc.ticker = clock.Ticker(time.Millisecond * time.Duration(intervalMillis))
 	svc.backgroundWorkers.Add(1)
 	goutils.PanicCapturingGo(func() {
 		defer svc.backgroundWorkers.Done()
-		defer ticker.Stop()
+		defer svc.ticker.Stop()
```

Comment:
Is calling svc.ticker.Stop() safe? Just making sure since this is being deferred and also called explicitly elsewhere